.github/workflows/CCOA.yml

```diff
@@ -0,0 +1,69 @@
+name: CCOA
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+
+permissions:
+  pull-requests: write
+
+jobs:
+  add_label_and_comment:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check current date securely
+        id: date_check
+        run: |
+          # Define the block date (Jan 7, 2025)
+          BLOCK_DATE=$(date -d "2025-01-07" +%s)
+
+          # Get the current date in seconds
+          CURRENT_DATE=$(date +%s)
+
+          # Validate the dates and safely output the result
+          if [ "$CURRENT_DATE" -lt "$BLOCK_DATE" ]; then
+            echo "continue=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "continue=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Add do-not-merge label
+        if: steps.date_check.outputs.continue == 'true'
+        uses: actions-ecosystem/action-add-labels@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          labels: do-not-merge
```

Comment:
`actions-ecosystem/action-add-labels` is a third-party action, how about using `github.rest.issues.addLabels` in `actions/github-script`?